internal/pointer/pointer_test.go

```diff
@@ -0,0 +1,51 @@
+package pointer
+
+import (
+	"testing"
+)
+
+type testCase[T comparable] struct {
+	name         string
+	val          T
+	defaultValue T
+	wantNil      bool
+}
+
+func assertSetOrNil[T comparable](t *testing.T, tc testCase[T]) {
+	t.Run(tc.name, func(t *testing.T) {
+		ptr := SetOrNil(tc.val, tc.defaultValue)
+		if gotNil := ptr == nil; gotNil != tc.wantNil {
+			t.Errorf("got nil %t, want %t", gotNil, tc.wantNil)
+		}
+	})
+}
+
+func TestSetOrNil(t *testing.T) {
+	assertSetOrNil(t, testCase[int]{
```

Comment:
nit pick, why not putting those in a table, use a loop and use the name as a string for a **t.Run**

Reply:
because it uses generics which I cannot table test for different types